support/packages/local-npm/README.md

```diff
@@ -0,0 +1,213 @@
+
+The `local-npm` has three commands, of which you will normally use two:
+
+1. `install`: this works like `yarn install` but it makes sure that the package
+   is re-downloaded from the repo and written again to your `node_modules`
+   folder. On the contrary, `yarn` doesn't do this unless you change the
+   dependency version number or remove `yarn.lock` and `node_modules`.
+2. `publish`: this is like `npm publish` but it makes sure you are publishing to
+   your local repo only and overwrites any published package with the same name
+   and version.
+3. `registry`: this has two subcommands for querying (`get`) the active
+   repository, or setting (`set`) it to `local` or `public`.
+
+Usually you will always have your `npm` and `yarn` repos pointing to your local
+repo. This is because Verdaccio acts as a proxy of the real public repo so, if
+you request anything it doesn't have, it will download it, cache it and serve
+it.
+
+This has the benefit that you don't need to switch between local and public
+repos whenever you use third party packages and, also, that you will have a
+local mirror of the public `npm` repo which will let you develop even when
+offline (this is not a wanted feature of the setup, but a nice side effect).
+
+The only moments when you need to switch to using the public repos are:
+
+1. If you are going to update a `yarn.lock` file that you will then commit to a
+   GitHub repo.
+2. If you want to make sure that a public release has gone well and thus you
+   need the possibility that your local repo serves packages that are not really
+   available to the public.
+
+#### Case 1
+
+Case 1 will usually happen at the end of a pull request preparation. Whenever we
+change deps in `liferay-frontend-projects`, we must run `yarn install` so that
+`yarn.lock` is updated and we can commit the modified version.
+
+However, `yarn.lock` files hold information about the repo from where the
+packages are downloaded so, if you run `yarn install` when your machine is
+pointing to your local repo, the `yarn.lock` files will have references to
+`localhost` instead of `registry.npmjs.org`, and those will fail in other
+machines.
+
+For that reason, the recommended action to take when you finish preparing your
+pull request are:
+
+1. Run `git checkout yarn.lock` to restore it to `master`'s state
+2. Run `local-npm registry set public` to point `npm` and `yarn` to the public
+   repo.
+3. Run `yarn install` to update `yarn.lock`
+4. Run `local-npm registry set local` to restore your repos back again to use
+   your local Verdaccio.
+
+In the event that you forget to follow these steps or commit a `yarn.lock` file
+with references to `localhost` there's no need for drama (though you can make
+drama of it if you like; it's OK) because our CI build will catch the error and
+tell you that your commited `yarn.lock` is incorrect.
+
+In that case, simply fix it, force push the `yarn.lock` commit, and continue
+with you beautiful life.
```

Comment:
```suggestion
Case 1 will usually happen at the end of a pull request preparation. Whenever we
change dependencies in `liferay-frontend-projects`, we must run `yarn install` so that
`yarn.lock` is updated and we can commit the modified version.

However, `yarn.lock` files hold information about the registry from where the
packages are downloaded. If you run `yarn install` when your machine is
pointing to your local registry, the `yarn.lock` files will have references to
`localhost` instead of `registry.npmjs.org`, and those will fail in other
machines.

For that reason, the recommended action to take when you finish preparing your
pull request are:

1. Run `git checkout yarn.lock` to restore it to `master`'s state
2. Run `local-npm registry set public` to point `npm` and `yarn` to the public
   repo.
3. Run `yarn install` to update `yarn.lock`
4. Run `local-npm registry set local` to restore your repos back again to use
   your local registry.

In the event that you forget to follow these steps or commit a `yarn.lock` file
with references to `localhost` there's no need for drama (though you can make
drama of it if you like; it's OK) because our CI build will catch the error and
tell you that your commited `yarn.lock` is incorrect.

In that case, simply fix it, force push the `yarn.lock` commit, and continue
with you beautiful life.
```